tools/scripts/mcufla.sh

```diff
@@ -109,10 +123,47 @@ build_prerequisites() {
 	fi
 }
 
-parse_input $@
+build_prerequisites_stm32() {
+	local HALF_CORES=$(expr $(grep -c '^processor' /proc/cpuinfo) / 2)
+
+	if [ ! -d ~/.stm32/openocd ]; then
+		git clone https://github.com/STMicroelectronics/OpenOCD.git ~/.stm32/openocd &&
+		pushd ~/.stm32/openocd &&
+		./bootstrap &&
+		./configure &&
+		make SHELL='sh' -j$HALF_CORES &&
+		popd
 
-build_prerequisites
+		if [ $? -ne 0 ]; then
+			rm -rf ~/.stm32/openocd
+			exit
+		fi
+	fi
+}
 
-print_run ~/.maxflash/openocd/src/openocd -s ~/.maxflash/openocd/tcl \
-	  -c "adapter driver cmsis-dap; transport select swd; $DAP_SERIAL" \
-	  -f "target/$TARGET.cfg"  -c "program $1 verify reset exit"
+configure_specific_platform() {
+	if [ $PLATFORM == 'maxim' ]; then
+		echo "Maxim platform detected"
+		build_prerequisites_maxim
+		print_run ~/.maxflash/openocd/src/openocd -s ~/.maxflash/openocd/tcl \
+			-c "adapter driver cmsis-dap; transport select swd; $DAP_SERIAL" \
+			-f "target/$TARGET.cfg"  -c "program $1 verify reset exit"
+	elif [ $PLATFORM == 'stm32' ]; then
+		echo "STM32 platform detected"
+		TAR="$TARGET"x.cfg
+		build_prerequisites_stm32
+		print_run ~/.stm32/openocd/src/openocd -s ~/.stm32/openocd/tcl \
+			-f interface/stlink-dap.cfg -c "transport select dapdirect_swd; $DAP_SERIAL" \
+			-f "target/$TAR" -c "program $1 verify reset exit"
+	elif [ $PLATFORM == 'pico' ]; then
+		echo "Pico platform detected"
+		print_run openocd -f interface/cmsis-dap.cfg -f "target/$TARGET" \
```

Comment:
What version of openocd does pico work with. Shouldn't it also be installed similar to the other platforms?

Reply:
Rpi pico devs managed to upstream their configuration files and so we can use stock openocd.

Apparently it even works with the ST fork, probably they upstreamed it long ago or ST devs rebase often (I didn't check).